File: leetcode/1877/solution.py
"""
https://leetcode.com/problems/minimize-maximum-pair-sum-in-array/
1977. Minimize Maximum Pair Sum in Array
"""


class Solution:
    def minPairSum(self, nums: List[int]) -> int:
        nums.sort()
        return max([nums[i] + nums[-i - 1] for i in range(len(nums) // 2)])

File: leetcode/1877/solution.rs
/**
 * https://leetcode.com/problems/minimize-maximum-pair-sum-in-array/
 * 1877. Minimize Maximum Pair Sum in Array
 */
impl Solution {
    pub fn min_pair_sum(mut nums: Vec<i32>) -> i32 {
        nums.sort_unstable();
        let l = nums.len() - 1;
        nums.iter()
            .enumerate()
            .fold(0, |acc, (i, n)| acc.max(n + nums[l - i]))
    }
}
